DatadogInternal/Sources/NetworkInstrumentation/W3C/W3CHTTPHeadersWriter.swift

```diff
@@ -61,48 +61,52 @@ public class W3CHTTPHeadersWriter: TracePropagationHeadersWriter {
         typealias Constants = W3CHTTPHeaders.Constants
 
         let sampled = traceContext.isKept
+        let shouldInject: Bool = {
+            switch traceContextInjection {
+                case .all:      return true
+                case .sampled:  return sampled
+            }
+        }()
+        guard shouldInject else {
+            return
+        }
 
-        switch (traceContextInjection, sampled) {
-        case (.all, _), (.sampled, true):
```

Comment:
Same here. Behaviour was correct according to tests.

I hope it's a bit less cryptic now.